ffi/src/lib.rs

```diff
@@ -928,6 +938,60 @@ unsafe fn open_db(args: &CreateOrOpenArgs) -> Result<Db, String> {
     Db::new_sync(path, cfg).map_err(|e| e.to_string())
 }
 
+/// Start logs for this process.
+///
+/// # Arguments
+///
+/// See `LogsArgs`.
+///
+/// # Returns
+///
+/// A `Value` containing {0, null} if the global logger was initialized.
+/// A `Value` containing {0, "error message"} if an error occurs.
+#[unsafe(no_mangle)]
+pub extern "C" fn fwd_start_logs(args: &LogArgs) -> Value {
+    start_logs(args).map_or_else(Into::into, Into::into)
+}
+
+#[doc(hidden)]
+fn start_logs(log_args: &LogArgs) -> Result<(), String> {
+    let cstr = unsafe { CStr::from_ptr(log_args.path) };
+    let osstr = OsStr::from_bytes(cstr.to_bytes());
+    let log_path = if osstr.is_empty() {
+        Path::join(&std::env::temp_dir(), "firewood-log.txt")
+    } else {
+        PathBuf::from(osstr)
+    };
+
+    let log_dir = Path::new(&log_path)
+        .parent()
+        .ok_or("failed to get log directory")?;
+    std::fs::create_dir_all(log_dir).map_err(|e| e.to_string())?;
+
+    let cstr = unsafe { CStr::from_ptr(log_args.filter_level) };
+    let level_str = OsStr::from_bytes(cstr.to_bytes())
+        .to_str()
+        .unwrap_or("info");
+    let level = level_str
+        .parse::<log::LevelFilter>()
+        .map_err(|e| e.to_string())?;
```

Comment:
Same here, you must first check that `filter_level` is not null before calling `CStr::from_ptr` otherwise this will cause undefined behavior.